### DISCUSSION TITLE
openclash开启后，手机可以顺利打开谷歌商店，但无法下载APP，显示等待中，并且还有幽灵事件！！！

### DISCUSSION THREAD

**Discussion by wdjkgit** [vernesong/OpenClash#3131] [Q&A]
这几天一直为手机无法从谷歌商店下载APP折腾，先后把openclash更新到最新的V0.45.103版本，并且在自定义规则中也填加了- DOMAIN-SUFFIX,googleapis.cn,🇹🇼 台湾节点    - DOMAIN-SUFFIX,xn--ngstr-lra8j.com,🇹🇼 台湾节点  ，但情况依旧，也先后试了网站的clash托管订阅和SSR订阅。但至今没有解决。
  但奇怪的是下载APP时，显示等待中时，如果重启openwrt时或者手机关闭连接到openwrt路由的wifi网络，却可以短暂的下载数据。目前就是在反复的切换网络中完成下载任务。。请高手解决！！！

**Comment by ehu1084** [comment]
同，逼得我只能再手机上开clash下载app

**Reply by wdjkgit** [reply]
目前发现可能是DNS的问题，我现在关闭了openclash的本地 DNS 劫持，开放了7874端口给adguard home去广告，去广告的效果目前没有发现有什么效果，但谷歌商店正常了！ @

**Comment by AoThen** [comment]
![image](https://github.com/vernesong/OpenClash/assets/32790381/e8a3f3df-e6ca-4910-b8c9-0b015abd7115)

```
##解决绕过大陆后谷歌商店无法更新
services.googleapis.cn
googleapis.cn
```
其实作者有提到过解决方法，给大家标识一下吧，最终解决方法 @wdjkgit @ehu1084

**Reply by JefferWong** [reply]
这个默认是没有注释掉的。似乎问题没有解决。

**Reply by m324216** [reply]
> services.googleapis.cn
> googleapis.cn

亲测已经解决，感谢！

**Reply by supersnsd** [reply]
> ![image](https://private-user-images.githubusercontent.com/32790381/293442689-e8a3f3df-e6ca-4910-b8c9-0b015abd7115.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgwMjQ1ODcsIm5iZiI6MTcxODAyNDI4NywicGF0aCI6Ii8zMjc5MDM4MS8yOTM0NDI2ODktZThhM2YzZGYtZTZjYS00OTEwLWI4YzktMGIwMTVhYmQ3MTE1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEwVDEyNTgwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVhYWM5ODZhM2QzYWFkMTNiM2JkODhiZDQ3ZmU5N2Y0YzUzZDI4ZTNmMDY4ODE4ODQ5ZTBlOGRmZGFhYmQ5ZjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.sIOiTVwv896cXDMY-sRnvGv-Qv-3W1JfKcvJ3FYf8Ao)
> 
> ```
> ##解决绕过大陆后谷歌商店无法更新
> services.googleapis.cn
> googleapis.cn
> ```
> 
> 其实作者有提到过解决方法，给大家标识一下吧，最终解决方法 @wdjkgit @ehu1084

失效了,现在怎么设置都没用

**Comment by jaxluoqs** [comment]
在配置文件中把xn--ngstr-lra8j.com 设为直连

**Reply by zent1n0** [reply]
像是订阅转换的问题 不知道为什么把这个suffix加到代理上了

**Reply by SkywalkerSpace** [reply]
亲测可用。我用的 rule-set 把这个域名代理了，改为直连就能下载 app 了

**Reply by jisoo163** [reply]
亲测可用 牛

**Reply by yegetables** [reply]
好用

**Comment by 2315095914** [comment]
加个ip-cidr代理了就有用了，我现在的是- IP-CIDR,120.253.253.98/32,Proxy 不知道能持续多久

**Comment by ATa0ist** [comment]
我一两年之前的解决办法，一直也没出问题，把这几个加到绕过大陆黑名单好就好了
![Snipaste_2024-08-19_15-24-21](https://github.com/user-attachments/assets/c5551bc5-003c-4600-b159-b9de67abeefc)

**Reply by 18115005** [reply]
解决了，谢谢

**Reply by weicaixian86** [reply]
> 解决了，谢谢

是怎么解决的。？？？？

**Reply by zent1n0** [reply]
> 在配置文件中把xn--ngstr-lra8j.com 设为直连

这个就能简单解决 压根不用什么DNS重写或者adguard home

**Reply by shouko17** [reply]
解决了,感谢

**Comment by weicaixian86** [comment]
您好，您发送的邮件已经收到。

**Comment by Rangtian** [comment]
我把 googleapis.cn 添加到 插件设置 -> 流量控制 -> 绕过指定区域 IPv4 黑名单，就能正常下载、更新App了。
openclash 我是按照[不良林的教程](https://www.youtube.com/watch?v=s84CWgKus4U)设置，避免了黑名单网站的 DNS 泄露，同时开启了流量嗅探。

![微信截图_20240928115555](https://github.com/user-attachments/assets/60413613-96e2-4e54-bc1b-a137fa0865bc)

**Reply by zent1n0** [reply]
openclash有订阅转换吧 转换的时候谷歌商店的两条相关规则路由不一致就会有这个问题

**Reply by qq625970512** [reply]
> 我把 googleapis.cn 添加到 插件设置 -> 流量控制 -> 绕过指定区域 IPv4 黑名单，就能正常下载、更新App了。 openclash 我是按照[不良林的教程](https://www.youtube.com/watch?v=s84CWgKus4U)设置，避免了黑名单网站的 DNS 泄露，同时开启了流量嗅探。
> 
> ![微信截图_20240928115555](https://private-user-images.githubusercontent.com/22640407/371723495-60413613-96e2-4e54-bc1b-a137fa0865bc.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjM1MjMwMzMsIm5iZiI6MTc2MzUyMjczMywicGF0aCI6Ii8yMjY0MDQwNy8zNzE3MjM0OTUtNjA0MTM2MTMtOTZlMi00ZTU0LWJjMWItYTEzN2ZhMDg2NWJjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTE5VDAzMjUzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM5Yjk4N2E1ODkyYzIyODRmMWI4YTFkNTIzNmUyMjY2MGQxNjgwNDc0MDk2OWFmMGVlMzEzMzUxNTY3MmViMzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.JE_sPgr6qFxGFh2q1zc52Xv_1h2BSa9u3J0vDQdBNfg)

我添加了,但是更新到0.47后的版本后又失效了,不知道怎么办了,以前的0.46版本时候是有效果的

**Comment by weicaixian86** [comment]
您好，您发送的邮件已经收到。

**Comment by LiJiaHua1024** [comment]
参考[https://www.yetpage.com/archives/278#toc-head-15](https://www.yetpage.com/archives/278#toc-head-15)

> "services.googleapis.cn"解析为国内地址时，会返回国内cdn的下载地址，其域名形式为"[xxxx]-xn--ngstr-lra8j.com",而且国内cdn会阻止国外ip的访问

所以在`绕过指定区域 IPv4 黑名单`中，我是这么设置的：
```
services.googleapis.cn
googleapis.cn
*xn--ngstr-lra8j.com
clientservices.googleapis.com
googleapis.cn
```
之前写的`xn--ngstr-lra8j.com`没有用，发现好像可以用通配符，改成`*xn--ngstr-lra8j.com`就可以正常下载了

**Reply by zfdang** [reply]
改了这个之后就好使了，赞！

**Reply by hang1233** [reply]
> 参考https://www.yetpage.com/archives/278#toc-head-15
> 
> > "services.googleapis.cn"解析为国内地址时，会返回国内cdn的下载地址，其域名形式为"[xxxx]-xn--ngstr-lra8j.com",而且国内cdn会阻止国外ip的访问
> 
> 所以在`绕过指定区域 IPv4 黑名单`中，我是这么设置的：
> 
> ```
> services.googleapis.cn
> googleapis.cn
> *xn--ngstr-lra8j.com
> clientservices.googleapis.com
> googleapis.cn
> ```
> 
> 之前写的`xn--ngstr-lra8j.com`没有用，发现好像可以用通配符，改成`*xn--ngstr-lra8j.com`就可以正常下载了

谢谢您

**Reply by qq625970512** [reply]
![111111](https://github.com/user-attachments/assets/b0d8f1fa-e262-4adc-950e-a3ed1f4b810b)
不知道为何,并没有效果,版本是0.47,是应为我填写了2条[xn--ngstr-lra8j.com](https://xn--ngstr-lra8j.com/)的关系吗?但是实际我去掉过也没用

**Reply by chenshuangruoyi** [reply]
你把之前注释的前面三条删除掉只保留后面的8条就可以解决问题了，之前我也是搞了好久今天尝试删除前面三条保存配置然后应用配置，打开手机的谷歌商店神奇的可以下载更新了

**Reply by qq625970512** [reply]
> 你把之前注释的前面三条删除掉只保留后面的8条就可以解决问题了，之前我也是搞了好久今天尝试删除前面三条保存配置然后应用配置，打开手机的谷歌商店神奇的可以下载更新了

前三条是注释,而已啊,又不是实际生效的,我测试的时候修改过参数后的确短期内能下载,但是时间一长或者重启后就失效了,麻烦你那边也试试看

**Reply by supersnsd** [reply]
解决了么？

**Reply by chenshuangruoyi** [reply]
> > 你把之前注释的前面三条删除掉只保留后面的8条就可以解决问题了，之前我也是搞了好久今天尝试删除前面三条保存配置然后应用配置，打开手机的谷歌商店神奇的可以下载更新了
> 
> 前三条是注释,而已啊,又不是实际生效的,我测试的时候修改过参数后的确短期内能下载,但是时间一长或者重启后就失效了,麻烦你那边也试试看

我就是这样设置的，用了个把月了都没有问题啊，而且我的软路由设置的每个月1号11号21号自动重启，没有像你说的会失效，今天打开了谷歌商店很多程序昨天就已经自动更新了

**Reply by qq625970512** [reply]
> > > 你把之前注释的前面三条删除掉只保留后面的8条就可以解决问题了，之前我也是搞了好久今天尝试删除前面三条保存配置然后应用配置，打开手机的谷歌商店神奇的可以下载更新了
> > 
> > 
> > 前三条是注释,而已啊,又不是实际生效的,我测试的时候修改过参数后的确短期内能下载,但是时间一长或者重启后就失效了,麻烦你那边也试试看
> 
> 我就是这样设置的，用了个把月了都没有问题啊，而且我的软路由设置的每个月1号11号21号自动重启，没有像你说的会失效，今天打开了谷歌商店很多程序昨天就已经自动更新了

我这里没办法,原因不明,
<img width="721" height="403" alt="image" src="https://github.com/user-attachments/assets/83e2718c-0b6c-4d66-8314-9d425cbfb65e" />
已经按照你说的,去掉了上面三条注释后,实际测试还是谷歌商店无法下载,就是提示等待中

**Reply by qq625970512** [reply]
> > > 你把之前注释的前面三条删除掉只保留后面的8条就可以解决问题了，之前我也是搞了好久今天尝试删除前面三条保存配置然后应用配置，打开手机的谷歌商店神奇的可以下载更新了
> > 
> > 
> > 前三条是注释,而已啊,又不是实际生效的,我测试的时候修改过参数后的确短期内能下载,但是时间一长或者重启后就失效了,麻烦你那边也试试看
> 
> 我就是这样设置的，用了个把月了都没有问题啊，而且我的软路由设置的每个月1号11号21号自动重启，没有像你说的会失效，今天打开了谷歌商店很多程序昨天就已经自动更新了

- DOMAIN-SUFFIX,xn--ngstr-lra8j.com,DIRECT
- DOMAIN-SUFFIX,googleapis.cn,DIRECT
- DOMAIN,clientservices.googleapis.com,DIRECT
- DOMAIN,update.googleapis.com,DIRECT
##- DOMAIN,translate.googleapis.com,DIRECT #屏蔽后让谷歌翻译能使用
- DOMAIN,fonts.googleapis.com,DIRECT
这是我唯一能正常下载的方法,在覆写里的自定义规则里添加后才能使用,其他方式都没用,都是能打开,但是无法下载

**Reply by supersnsd** [reply]
> > > > 你把之前注释的前面三条删除掉只保留后面的8条就可以解决问题了，之前我也是搞了好久今天尝试删除前面三条保存配置然后应用配置，打开手机的谷歌商店神奇的可以下载更新了
> > > 
> > > 
> > > 前三条是注释,而已啊,又不是实际生效的,我测试的时候修改过参数后的确短期内能下载,但是时间一长或者重启后就失效了,麻烦你那边也试试看
> > 
> > 
> > 我就是这样设置的，用了个把月了都没有问题啊，而且我的软路由设置的每个月1号11号21号自动重启，没有像你说的会失效，今天打开了谷歌商店很多程序昨天就已经自动更新了
> 
> * DOMAIN-SUFFIX,xn--ngstr-lra8j.com,DIRECT
> * DOMAIN-SUFFIX,googleapis.cn,DIRECT
> * DOMAIN,clientservices.googleapis.com,DIRECT
> * DOMAIN,update.googleapis.com,DIRECT
>   ##- DOMAIN,[translate.googleapis.com](https://translate.googleapis.com),DIRECT #屏蔽后让谷歌翻译能使用
> * DOMAIN,fonts.googleapis.com,DIRECT
>   这是我唯一能正常下载的方法,在覆写里的自定义规则里添加后才能使用,其他方式都没用,都是能打开,但是无法下载

我还是不行，可以下载，但无法下完，就卡后面百分之几不动，停了再下都是如此

**Reply by chenshuangruoyi** [reply]
> > > > 你把前面注释的前面三条删除只保留后面的8条就可以解决问题了，之前我也搞了好久今天尝试删除前面三条保存配置然后应用配置，打开手机的谷歌商店神奇的可以下载更新了
> > > 
> > > 
> > > 前三条是注释​​，最后啊，又不是实际生效的，我测试的时候修改一下参数后短期内能下载，但是时间一长重启或者后面就失效了，麻烦你这边也试试看
> > 
> > 
> > 我就是这样设置的，用了个把月了都没有问题啊，而且我的软路由设置的每个月1号11号21号自动重启，没有像你说的会失效，今天打开了谷歌商店很多程序昨天就已经自动更新了
> 
> * 域名后缀,xn--ngstr-lra8j.com,直接
> * 域名后缀，googleapis.cn，直接
> * 域名,clientservices.googleapis.com,直接
> * DOMAIN,update.googleapis.com,DIRECT 
>   ##- DOMAIN,translate.googleapis.com,DIRECT #发光后让谷歌翻译能使用
> * DOMAIN,fonts.googleapis.com,DIRECT
>   这是我唯一能正常下载的方法，在覆写里的自定义规则里添加后才能使用，其他方式都用不了，都可以打开，但是无法下载

我是这样粘贴的，你可以试一试，还是不行可能你别的配置有冲突，会不会是你自己更改了别的配置，还有就是节点可能有问题，我用的是谷歌云搭建的，目前还蛮稳定的，在电视上看4K视频还行。
![Screenshot_2026-03-18-21-49-37-419_com android chrome](https://github.com/user-attachments/assets/c5bc03ba-cb09-4a0f-9bf3-a154cd100cca)

**Reply by chenshuangruoyi** [reply]
> > > > > 你把前面注释的前面三条删除只保留后面的8条就可以解决问题了，之前我也搞了好久今天尝试删除前面三条保存配置然后应用配置，打开手机的谷歌商店神奇的可以下载更新了
> > > > 
> > > > 
> > > > 前三条是注释​​，最后啊，又不是实际生效的，我测试的时候修改一下参数后短期内能下载，但是时间一长重启或者后面就失效了，麻烦你这边也试试看
> > > 
> > > 
> > > 我就是这样设置的，用了个把月了都没有问题啊，而且我的软路由设置的每个月1号11号21号自动重启，没有像你说的会失效，今天打开了谷歌商店很多程序昨天就已经自动更新了
> > 
> > 
> > 
> > * 域名后缀,xn--ngstr-lra8j.com,直接
> > * 域名后缀，googleapis.cn，直接
> > * 域名,clientservices.googleapis.com,直接
> > * DOMAIN,update.googleapis.com,DIRECT 
> >   ##- DOMAIN, [translate.googleapis.com](https://translate.googleapis.com) ,DIRECT #发光后让谷歌翻译能使用
> > * DOMAIN,fonts.googleapis.com,DIRECT
> >   这是我唯一能正常下载的方法，在覆写里的自定义规则里添加后才能使用，其他方式都用不了，都可以打开，但是无法下载
> 
> 我还是不行，可以下载，但无法下完，就卡后面百分之几不动，停了再下都是如此

有时候我也会碰到下载到百分之九十几就不动了的，我也没管他，估计还是配置设置的有问题，这个排除的因素蛮多的

**Reply by supersnsd** [reply]
是的，用cf做的订阅不行，油管外网gpt都行，就是商店下载不行，搜完全网没找到解决方法...目前是换其他收费机场节点才行，应该是某些没走代理或不通

**Reply by hhht110** [reply]
> 是的，用cf做的订阅不行，油管外网gpt都行，就是商店下载不行，搜完全网没找到解决方法...目前是换其他收费机场节点才行，应该是某些没走代理或不通

对，个别机场也有这样的情况，乱屏蔽，连分流，我目前用良心云没遇到过这样的情况，我啥都没设置，就直接打开openclash就能用，  不是打广告哈， 反正便宜还好用， 还有的会屏蔽个别网站，挺烦的

**Reply by qq625970512** [reply]
> > > > > 你把前面注释的前面三条删除只保留后面的8条就可以解决问题了，之前我也搞了好久今天尝试删除前面三条保存配置然后应用配置，打开手机的谷歌商店神奇的可以下载更新了
> > > > 
> > > > 
> > > > 前三条是注释​​，最后啊，又不是实际生效的，我测试的时候修改一下参数后短期内能下载，但是时间一长重启或者后面就失效了，麻烦你这边也试试看
> > > 
> > > 
> > > 我就是这样设置的，用了个把月了都没有问题啊，而且我的软路由设置的每个月1号11号21号自动重启，没有像你说的会失效，今天打开了谷歌商店很多程序昨天就已经自动更新了
> > 
> > 
> > 
> > * 域名后缀,xn--ngstr-lra8j.com,直接
> > * 域名后缀，googleapis.cn，直接
> > * 域名,clientservices.googleapis.com,直接
> > * DOMAIN,update.googleapis.com,DIRECT
> >   ##- DOMAIN,translate.googleapis.com,DIRECT #发光后让谷歌翻译能使用
> > * DOMAIN,fonts.googleapis.com,DIRECT
> >   这是我唯一能正常下载的方法，在覆写里的自定义规则里添加后才能使用，其他方式都用不了，都可以打开，但是无法下载
> 
> 我是这样粘贴的，你可以试一试，还是不行可能你别的配置有冲突，会不会是你自己更改了别的配置，还有就是节点可能有问题，我用的是谷歌云搭建的，目前还蛮稳定的，在电视上看4K视频还行。 ![Screenshot_2026-03-18-21-49-37-419_com android chrome](https://private-user-images.githubusercontent.com/191772762/565592720-c5bc03ba-cb09-4a0f-9bf3-a154cd100cca.jpg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzM5MDg0NjcsIm5iZiI6MTc3MzkwODE2NywicGF0aCI6Ii8xOTE3NzI3NjIvNTY1NTkyNzIwLWM1YmMwM2JhLWNiMDktNGEwZi05YmYzLWExNTRjZDEwMGNjYS5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMzE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDMxOVQwODE2MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZWI0ZmMyMzJmZWU1YTE0N2M3NGJiZjU5NjZjODIzMmE5MGFhNTY2NzY2NDE2ZmMzNWYxYmY2ZTFhZDdhMDBiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.lQiSW5UovxDYc3lNfI6fohDBMJbWH214vt5rrd_lhvs)

当我更新到v0.47.055版本后,按照你的设定来使用,并且保留了上面3条注释依旧可以正常下载,我打算继续观察一下看看,很大的可能跟openclash版本可能有关系

**Reply by qq625970512** [reply]
> > 是的，用cf做的订阅不行，油管外网gpt都行，就是商店下载不行，搜完全网没找到解决方法...目前是换其他收费机场节点才行，应该是某些没走代理或不通
> 
> 对，个别机场也有这样的情况，乱屏蔽，连分流，我目前用良心云没遇到过这样的情况，我啥都没设置，就直接打开openclash就能用， 不是打广告哈， 反正便宜还好用， 还有的会屏蔽个别网站，挺烦的

请问能否提供下良心云的地址呢?我最近也在找新的机场做备用,我想去了解一下

**Comment by weicaixian86** [comment]
您好，您发送的邮件已经收到。

**Comment by Kente-Z** [comment]
我发现我开始下载的时候切换到流量就可以了，所以我的是services.googleapis.cn
googleapis.cn这两个域名走代理，然后xn--ngstr-lra8j.com直连，就可以了

**Comment by zent1n0** [comment]
全部相关域名设置代理的方法上面都介绍了 在这里提供一个能够让谷歌商店下载走国内CDN的思路：
将
```
- DOMAIN-SUFFIX,xn--ngstr-lra8j.com,DIRECT
- DOMAIN-SUFFIX,googleapis.cn,DIRECT
- DOMAIN,clientservices.googleapis.com,DIRECT
- DOMAIN,update.googleapis.com,DIRECT
- DOMAIN,translate.googleapis.com,DIRECT
- DOMAIN,fonts.googleapis.com,DIRECT
```
增加到规则头部 clientservices.googleapis.com走直连 最终使得[xxxx]-xn--ngstr-lra8j.com解析结果为国内CDN 更新应用不走节点
商店浏览所需的剩余api会和其他google的分流策略一致 根据机场提供的规则 一般是走到节点代理

注意不要使用`DOMAIN-SUFFIX,googleapis.cn,DIRECT`的分流方式 过于宽泛的前缀规则会导致商店的其余资源走直连无法加载

**Reply by qq625970512** [reply]
![QQ截图20251203095202](https://github.com/user-attachments/assets/6abc74ed-8f52-4f60-847f-62a1b6a8ed29)
我在这里已经添加过域名了,但是还是没效果,至于大佬您这里贴出来的规则我不懂怎么使用,现在菜鸟的我还是用的默认的配置,单纯订阅的时候使用了转换规则,其实准确的说在上半年的时候还是有效果的,还是能更新的,在配置不变的情况下,下半年开始就无法正常下载谷歌商店里的应用了,只能在手机上开软件后才能下载,光用路由器就不行了

**Reply by mickeyman115** [reply]
你更新软件时他会一直转圈不动是吧，这个时候关闭wifi使用手机流量，他就开始下了

**Reply by m1a2sep** [reply]
好人一生平安，折腾我好几天了，感谢您！

**Reply by zent1n0** [reply]
@qq625970512 自定义规则可以通过订阅转换实现 或者多数主流的clash客户端基本都有提供插入自定义规则的功能

具体到openclash 可以在覆写设置-规则设置中 勾选自定义规则 然后将你需要的自定义规则加进来

绕过指定区域IPv4黑名单功能 实际上通过tun 的fake-ip filter实现 如果小白不明白这个功能意味着什么 推荐还是尽量使用自定义规则来实现分流

**Reply by qq625970512** [reply]
> @qq625970512 自定义规则可以通过订阅转换实现 或者多数主流的clash客户端基本都有提供插入自定义规则的功能
> 
> 具体到openclash 可以在覆写设置-规则设置中 勾选自定义规则 然后将你需要的自定义规则加进来
> 
> 绕过指定区域IPv4黑名单功能 实际上通过tun 的fake-ip filter实现 如果小白不明白这个功能意味着什么 推荐还是尽量使用自定义规则来实现分流

感谢大佬指点,已经自己琢磨到设置的地方,覆写设置-规则设置中 勾选自定义规则这里了,至于黑名单着个我没有单独设置,纯粹用他默认的配置,我今天换新小米手机前也是小米手机,用默认的黑名单设置足够使用,没有异常,但是今年换了手机后唯独不行,只能在自定义规则地方添加大佬你的规则后才能正常使用,这一点我很疑惑,同样的小米国产手机,为啥今年才遇到问题,往年怎么都好的

**Reply by qq625970512** [reply]
> 全部相关域名设置代理的方法上面都介绍了 在这里提供一个能够让谷歌商店下载走国内CDN的思路： 将
> 
> ```
> - DOMAIN-SUFFIX,xn--ngstr-lra8j.com,DIRECT
> - DOMAIN-SUFFIX,googleapis.cn,DIRECT
> - DOMAIN,clientservices.googleapis.com,DIRECT
> - DOMAIN,update.googleapis.com,DIRECT
> - DOMAIN,translate.googleapis.com,DIRECT
> - DOMAIN,fonts.googleapis.com,DIRECT
> ```
> 
> 增加到规则头部 clientservices.googleapis.com走直连 最终使得[xxxx]-xn--ngstr-lra8j.com解析结果为国内CDN 更新应用不走节点 商店浏览所需的剩余api会和其他google的分流策略一致 根据机场提供的规则 一般是走到节点代理
> 
> 注意不要使用`DOMAIN-SUFFIX,googleapis.cn,DIRECT`的分流方式 过于宽泛的前缀规则会导致商店的其余资源走直连无法加载

发现一个新问题,用了你的规则后,谷歌浏览器的翻译功能失效了,好像连接不上服务器了,谷歌商店下载是没问题了,但是谷歌浏览器的翻译功能失效了,能否再看一下问题

**Reply by shadowxhero** [reply]
translate.googleapis.com不能走直连，这个去掉

**Comment by hhht110** [comment]
这个问题以前很早遇到过。后来没用安卓了就没遇到过了。顺便提一嘴，少用ADGhome那玩意我认为不是很好用。简单的dns分流就好了。国内用去广告基本等于作死。国内这些软件厂商的操行真的是一言难尽，我刚刚试了试在电脑的谷歌游戏下载没有任何问题，我是openclash配合MOSDNS 目前设置好了 速度相当快。国内淘宝西瓜视频啥的都一切正常

**Reply by supersnsd** [reply]
我是用的mosdns最近下不了了,都是卡不同的百分率下载进度,就是下不完

**Comment by ylesc-ylc** [comment]
如果有人和我一样关闭了openclash的dns劫持，使用AdGuard Home进行广告拦截，可以将一下内容添加到AdGuard Home的上游dns服务器中：
[/services.googleapis.cn/]8.8.4.4
[/googleapis.cn/]8.8.4.4
[/xn--ngstr-lra8j.com/]8.8.4.4
[/clientservices.googleapis.com/]8.8.4.4
[/googleapis.cn/]8.8.4.4

仅供参考（在我的环境下已经解决问题）
<img width="1282" height="393" alt="dns" src="https://github.com/user-attachments/assets/5f81c03e-41da-419c-bb8f-2a3db5178210" />

**Reply by supersnsd** [reply]
用的mosdns如何操作

**Comment by candy9876** [comment]
我也被这个问题困扰了很久。目前经过测试发现，很大概率是ip地址被google送中了，gemini也不能使用的。这里如果用Hy2协议，就出现play不能下载，gemini不能使用的情况。 如果用了其他vless协议之类的，又可以正常使用， 就是速度会比较慢。hy2协议是真的快，就是容易出问题。 我的理解，是不是用其他协议sni伪装了之类的，ip地址不会被谷歌识别，所以能正常访问。 反正hy2协议就肯定是不行的
Gemini                    NO
GoogleSearch              YES
Google Play Store         NO (Region: CN)
